sqlmesh/core/plan/definition.py

```diff
@@ -700,6 +714,13 @@ def _refresh_dag_and_ignored_snapshots(self) -> None:
                 "Revert or apply changes before proceeding with restatements."
             )
 
+        if self._input_backfill_models is not None:
+            if not self.is_dev:
+                raise PlanError(
+                    "Selecting models to backfill is only supported for development environments."
+                )
+            self._models_to_backfill = set(self.__dag.subdag(*self._input_backfill_models).sorted)
```

Comment:
so this will select all models upstream of models selected to ensure the chain is proper right

Reply:
this is right